src/query_backend.h

```diff
@@ -7,19 +7,22 @@
 #include <set>
 #include <boost/version.hpp>
 
-#if BOOST_VERSION / 100 % 1000 <= 67
-  #include <boost/uuid/sha1.hpp>
-#else
-  #include <boost/uuid/detail/sha1.hpp>
-#endif
+#include <boost/uuid/detail/sha1.hpp>
 
 #include "blob.h"
 #include "rec_backend.h"
 #include "any.hpp"
 
-#define CYCLUS_UUID_SIZE 16
+#define CYCLUS_SHA1_NINT 5 // always treat the digest as if it were uint[5]
 #define CYCLUS_SHA1_SIZE 20
-#define CYCLUS_SHA1_NINT 5
+#define CYCLUS_UUID_SIZE 16
+#if BOOST_VERSION_MINOR < 86
+  #define BOOST_DIGEST_NINT 5
+  using BOOST_DIGEST_TYPE = unsigned int;
+#else
+  #define BOOST_DIGEST_NINT 20
+  using BOOST_DIGEST_TYPE = unsigned char;
+#endif
```

Comment:
Given how I see this new variable used, can't we just always use the 20 x char version, and always unpack it the way you have implemented it?

Reply:
Could also do some version of
`BOOST_DIGEST_NINT = CYCLUS_SHA1_NINT * sizeof(int)`?

Reply:
and maybe localize it to where it is used?

Reply:
> can't we just always use the 20 x char version, and always unpack it the way you have implemented it?

Boost < 1.86 still expects to store the digest as a 5 x uint, so this would fail to compile on older versions boost

> and maybe localize it to where it is used?

I localized the `#if / #else` to `Digest digest()`, and dropped the type aliases altogether.  I also changed `Digest` to a std::array, per the comment above `class Digest`.